### DISCUSSION TITLE
ement spellchecker

### DISCUSSION THREAD

**Discussion by OrionRandD** [alphapapa/ement.el#306] [Q&A]
Is there an Ement spellchecker to check messages before sending them to room(s)?

**Comment by alphapapa** [comment]
It's just Emacs.  You can use Flyspell or Jinx or whatever.

**Reply by OrionRandD** [reply]
Right. And how do I write my ID and password in, say, ~/authinfo.gpg file for emacs to access ement credenttials there? Instead of typing my ID an Pass every time I want to "ement-connect"?

**Reply by alphapapa** [reply]
Please see the customization options.

**Reply by OrionRandD** [reply]
I've seen the sso option, but anything related to "~/.authinfo.gpg" which, for me, is more user-friendly to Emacs authentication... I use it for mu4e and a bunch of other packages...

**Reply by phil-s** [reply]
The session file stores access tokens, not passwords.

The tokens are generated as a by-product of the authentication process.  You wouldn't store them in authinfo, as they can change (e.g. tokens can potentially expire).

**Reply by alphapapa** [reply]
> I've seen the sso option

What SSO option?

**Reply by OrionRandD** [reply]
> The session file stores access tokens, not passwords.
> 
> The tokens are generated as a by-product of the authentication process. You wouldn't store them in authinfo, as they can change (e.g. tokens can potentially expire).

Why not? I use mu4e in Emacs and my .authinfo.gpg is populated with app token to access my emails and do all sort of things with mu4e in Emacs...

**Reply by OrionRandD** [reply]
> > I've seen the sso option
> 
> What SSO option?

It opens firefox here and gives me four or five options to login through: github, google, et.al.; This is just what I do not want to giving access to these intrusive cursed companies...

**Reply by alphapapa** [reply]
> > > I've seen the sso option
> > 
> > 
> > What SSO option?
> 
> It opens firefox here and gives me four or five options to login through: github, google, et.al.; This is just what I do not want to giving access to these intrusive cursed companies...

Again, what SSO **option**?  I'm not aware of one existing.

If your server requires the use of SSO, that's up to the server admins.

**Reply by alphapapa** [reply]
> > The session file stores access tokens, not passwords.
> > The tokens are generated as a by-product of the authentication process. You wouldn't store them in authinfo, as they can change (e.g. tokens can potentially expire).
> 
> Why not? I use mu4e in Emacs and my .authinfo.gpg is populated with app token to access my emails and do all sort of things with mu4e in Emacs...

Because Emacs's `auth-source` library is not designed to be used with tokens that may change.  It's API is also simply broken in some ways.  I chose not to use it because doing so would result in users having to post issues like this asking how to work around it.  See https://old.reddit.com/r/emacs/comments/8lvda6/is_authsource_from_the_dark_side/
